xls/synthesis/timing_characterization_client_main.py

```diff
@@ -54,85 +88,304 @@ def _synth(stub: synthesis_service_pb2_grpc.SynthesisServiceStub,
-    kop: Operation name to emit into datapoints, generally in kConstant form for
-      use in the delay model; e.g. 'kAdd'.
-    stub: Handle to the synthesis server.
+  op = ENUM2NAME_MAP[kop]
 
-  Returns:
-    List of characterized datapoints via the synthesis server.
+  results = []
+  for bit_count in [1] + list(range(2, MAX_WIDTH, 2)):
+    op_type = f'bits[{bit_count}]'
+    ir_text = op_module_generator.generate_ir_package(op, op_type, (op_type, op_type))
+    module_name = f'{op}_{bit_count}'
+    mod_generator_result = op_module_generator.generate_verilog_module(module_name, ir_text)
+    top_name = module_name + '_wrapper'
+
+    verilog_text = op_module_generator.generate_parallel_module([mod_generator_result], top_name)
+    try:
+      result = _synth(stub, verilog_text, top_name)
+    except grpc.RpcError:
+      break
+
+    ps = 1e12/result.max_frequency_hz
+    result = delay_model_pb2.DataPoint()
+    result.operation.op = kop
+    result.operation.bit_count = bit_count
+    for i in range(2):
+      operand = result.operation.operands.add()
+      operand.bit_count = bit_count
+    result.delay = int(ps)
+    results.append(result)
+  model.data_points.extend(results)
+
+
+def _run_comparison(
+    kop: str, model: delay_model_pb2.DelayModel,
+    stub: synthesis_service_pb2_grpc.SynthesisServiceStub) -> None:
+  # Add op_model to protobuf message
+  add_op_model = model.op_models.add(op=kop)
+  add_op_model.estimator.regression.factors.add(
+    source=delay_model_pb2.DelayFactor.OPERAND_BIT_COUNT,
+    operand_number=0)
+
+  op = ENUM2NAME_MAP[kop]
 
-  Note that if a synthesis error occurs, it is assumed to be "out of resources"
-  and is quashed, with datapoints gathered to that point returned.
-  """
   results = []
-  for bit_count in [1] + list(range(2, 257, 2)):
+  for bit_count in [1] + list(range(2, MAX_WIDTH, 2)):
     op_type = f'bits[{bit_count}]'
-    ir_text = op_module_generator.generate_ir_package(op, op_type,
-                                                      (op_type, op_type))
+    ret_type = f'bits[1]'
+    ir_text = op_module_generator.generate_ir_package(op, ret_type, (op_type, op_type))
     module_name = f'{op}_{bit_count}'
-    mod_generator_result = op_module_generator.generate_verilog_module(
-        module_name, ir_text)
+    mod_generator_result = op_module_generator.generate_verilog_module(module_name, ir_text)
     top_name = module_name + '_wrapper'
-    verilog_text = op_module_generator.generate_parallel_module(
-        [mod_generator_result], top_name)
+    verilog_text = op_module_generator.generate_parallel_module([mod_generator_result], top_name)
     try:
       result = _synth(stub, verilog_text, top_name)
     except grpc.RpcError:
       break
 
-    ps = 1e12 / result.max_frequency_hz
+    ps = 1e12/result.max_frequency_hz
     result = delay_model_pb2.DataPoint()
     result.operation.op = kop
     result.operation.bit_count = bit_count
```

Comment:
not your bug, but:

result.operation.bit_count = 1

Reply:
(solved)